### PR TITLE
add method is_vertex_cut to (di)graphs

### PR DESCRIPTION
**Pull request by dcoudert**
This PR answers a query from https://ask.sagemath.org/question/78391/does-sage-have-a-function-to-determine-vertex-cuts/

It adds to (di)graphs a method to check if a set of vertices is a vertex cut of the (di)graph, that is if its removal from the (di)graph increases the number of (strongly) connected components. This generalizes method `is_cut_vertex`.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by lichengzhang1** [comment]
For edge versions, a similar problem involves adding a method is_edge_cut to (di)graphs.

**Comment by dcoudert** [comment]
> For edge versions, a similar problem involves adding a method is_edge_cut to (di)graphs.

Sure, but this should be done in another PR. This one is large enough.

**Comment by dcoudert** [comment]
@fchapoton I need some help here to rebase this branch. I tried to rebase on develop and fix merge conflicts, but  push was rejected
```
MAC-04017247:sage dcoudert$ git push origin HEAD:"graphs/is_vertex_cut"
To https://github.com/dcoudert/sage.git
 ! [rejected]                HEAD -> graphs/is_vertex_cut (non-fast-forward)
error: failed to push some refs to 'https://github.com/dcoudert/sage.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

**Comment by fchapoton** [comment]
say "thank you" to the recent doc patch-bombe..

I am not sure how to handle that properly ; my only safe strategy is the following :

did you try to click on the "resolve conflicts" button here and fix the conflicts online ?

**Comment by fchapoton** [comment]
- are you sure you have the latest develop ?
- you can use "git push -f" to force the push, but be cautious..

**Comment by dcoudert** [comment]
- I can't click on "resolve conflicts". It says that "These conflicts are too complex to resolve in the web editor"
- I have the last develop branch (according my clone on GitHub, and git pull)
- I don't see any "use the command line" button
- I will try a forced push...

**Comment by dcoudert** [comment]
Thanks @fchapoton, the forced push seems ok.

File: src/sage/graphs/connectivity.pyx [hunk 1]
```diff
@@ -904,68 +911,144 @@ def is_cut_vertex(G, u, weak=False):
...
 
     return False
 
 
+def is_cut_vertex(G, u, weak=False):
+    r"""
+    Check whether the input vertex is a cut-vertex.
+
+    A vertex is a cut-vertex if its removal from the (di)graph increases the
+    number of (strongly) connected components. Isolated vertices or leafs are
```
Comment (mkoeppe) [inline src/sage/graphs/connectivity.pyx]:
```suggestion
    number of (strongly) connected components. Isolated vertices or leaves are
```

Comment (dcoudert) [reply src/sage/graphs/connectivity.pyx]: Fixed

**Comment by dcoudert** [comment]
something went wrong with my last commit, trying to fix a merge conflict with 10.5.beta4.... I'm not sure how to fix that. 
Any advice is more than welcome.

**Comment by fchapoton** [comment]
well, github requires that the branch is **rebased** on top of the latest develop. If you merge instead, you git these hundreds of commits..

**Comment by fchapoton** [comment]
more precisely the commit message must look like
```Merge branch 'develop' into my_cool_branch``

**Comment by dcoudert** [comment]
I have restarted a branch on top of 10.5.beta4 with a forced push. It should be ok now.

**Comment by kwankyu** [comment]
```
    :meth:`is_cut_vertex` | Check whether the input vertex is a cut-vertex.
    :meth:`is_vertex_cut` | Check whether the input vertices form a vertex cut.
```
This is confusing to me. So `is_vertex_cut(G, [v])` does the same with `is_cut_vertex(G, v)`?

**Comment by kwankyu** [comment]
"cut vertex" seems a precise term, but "vertex cut" seems not. Do you agree?

**Comment by dcoudert** [comment]
1. Yes, `is_vertex_cut(G, [v])` does the same than `is_cut_vertex(G, v)`.
2. A set of edges whose removal from the graph increases the number of connected components is called a `edge cut` of the graph. Similarly, a set of vertices whose removal from the graph increases the number of connected components is called a `vertex cut`. So the terminology used here is the right one. When you refer to a `cut vertex` `v`, you refer to a single vertex whose removal increases the number of connected components. 

Now, we can ask if method `is_cut_vertex` is still needed when we have the more general method `is_vertex_cut`, and knowing that method `is_cut_vertex`  simply calls method `is_vertex_cut`. The same question holds for methods `is_cut_edge` and `is_edge_cut`.

**Comment by kwankyu** [comment]
OK. Thanks. Reasonable.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by kwankyu** [comment]
I will wait for ci checks.

**Comment by dcoudert** [comment]
The error reported by CI in `sage -t --long --random-seed=286735480429121101562228604801325644303 src/sage/schemes/elliptic_curves/hom_frobenius.py` seems unrelated to changes done in this PR

**Comment by kwankyu** [comment]
OK. Sorry that this PR got out of my radar.

**Comment by dcoudert** [comment]
Thank you for the review.

**Pull request merged**